core/templates/dev/head/services/AutogeneratedAudioPlayerService.js

```diff
@@ -24,9 +24,11 @@ oppia.factory('AutogeneratedAudioPlayerService', [
 
     var _speechSynthesis = window.speechSynthesis;
     // Not all browsers support SpeechSynthesisUtterance, so we need the
-    // fallback to prevent a "SpeechSynthesisUtterance is not defined" error.
-    var utterance = (
-      SpeechSynthesisUtterance ? new SpeechSynthesisUtterance() : null);
+    // check to prevent a "SpeechSynthesisUtterance is not defined" error.
+    var utterance = null;
+    if (window.hasOwnProperty('speechSynthesis')) {
```

Comment:
Would this be the same as just `if (_speechSynthesis)`, given line 25?

Reply:
Also would it be better to do something like this?

```
var _speechSynthesis = null;
var utterance = null;
if (window.hasOwnProperty('speechSynthesis')) {
  _speechSynthesis = window.speechSynthesis;
  utterance = new SpeechSynthesisUtterance();
}
```

That way you're not reading a property that might be undefined in line 25.

Reply:
Yeah, I moved it within the check.